docs/plugins.md

```diff
@@ -1 +1,124 @@
-TODO
\ No newline at end of file
+# TL;DR
+
+Similar to `kubectl` (the client CLI for Kubernetes) Kn now supports plugins. The plugin model for Kn was based on `kubectl` plugins but is different in some specific aspects—especially how plugins are executed and located on the user’s machine. In this document we give an overview of the Kn plugin model, highlight the differences with `kubectl` plugins and give some simple examples of Kn plugins in different languages to help you get started creating your own Kn plugins.
+
+## Introduction
+
+The Kn client aims to provide a command line interface (CLI) that is faithful to the Knative APIs. As such all Kn commands must have corresponding backend objects that they represent and expose to the user. There are a few commands that do not fit this philosophy, e.g., `help` and `completion`, however, these are the exceptions rather than the rule. 
+
+### Motivation
+
+This Kn design point raises various issues and questions such as: How to add esoteric or otherwise useful commands that users need for their workflows? How to add commands that are specific to Knative service providers? How to add meta-commands to install or upgrade and manage the Knative installation?
+
+To address these issues and questions and provide a general mechanism for extension, the Knative client team decided to add support for plugins to the Kn client. This mechanism solves similar problems as for the Kubernetes client (`kubectl`) and as such we have created Kn plugins to follow a similar architecture.
+
+### Model
+
+The `kubectl` plugin model is simple and quite powerful. In a nutshell, any executable placed in the user’s `PATH` whose name starts with prefix `kubectl-` is a plugin and will be invoked as part of `kubectl` itself. So an executable named `kubectl-hello` that is in the user’s `PATH` can be invoked with `kubectl` as:
```

Comment:
Why not immediately start with `kn` and use `kn` in the examples and then explain how `kubectl` is different to `kn` ? (if this is necessary at all). After all this is a documentation about `kn`, so let's explain how kubectl is different to kn instead of how kn is different to kubectl. Tbh, I would add this difference in a dedicated section at the end of this document.

Also its confusing to explain something here prominently and then say, well, for kn its different.